crates/matrix-sdk/src/event_cache/room/events.rs

```diff
@@ -241,27 +241,23 @@ impl RoomEvents {
 
     /// Remove some events from the linked chunk.
     ///
-    /// This method iterates over all event IDs in `event_ids` and removes the
-    /// associated event (if it exists) from `Self::chunks`.
-    pub fn remove_events_by_id(&mut self, event_ids: Vec<OwnedEventId>) {
-        for event_id in event_ids {
-            let Some(event_position) = self.revents().find_map(|(position, event)| {
-                (event.event_id().as_ref() == Some(&event_id)).then_some(position)
-            }) else {
-                error!(?event_id, "Cannot find the event to remove");
-
-                continue;
-            };
-
-            self.chunks
-                .remove_item_at(
-                    event_position,
-                    // If removing an event results in an empty chunk, the empty chunk is removed
-                    // because nothing is going to be inserted in it apparently.
-                    EmptyChunk::Remove,
-                )
-                .expect("Failed to remove an event we have just found");
+    /// Empty chunks are going to be removed.
+    ///
+    /// Events **must** be sorted by their position (descending, i.e. from
+    /// newest to oldest).
```

Comment:
This feels a bit footgun-y, can't we sort them here too, similar to what the deduplicator is doing in its `sort_events_by_position_descending`?